FirebaseCore/Internal/Sources/HeartbeatLogging/HeartbeatStorage.swift

```diff
@@ -51,20 +51,30 @@ final class HeartbeatStorage: HeartbeatStorageProtocol {
   // MARK: - Instance Management
 
   /// Statically allocated cache of `HeartbeatStorage` instances keyed by string IDs.
-  private static var cachedInstances: [String: WeakContainer<HeartbeatStorage>] = [:]
+  private nonisolated(unsafe) static var cachedInstances: [
+    String: WeakContainer<HeartbeatStorage>
+  ] =
+    [:]
+
+  /// Used to synchronize mutations
+  private static let instancesLock = NSLock()
```

Comment:
Should we add a todo about migrating to an actor when we get to iOS 13?

Reply:
I added an issue in the Firebase 12 milestone for tracking: https://github.com/firebase/firebase-ios-sdk/issues/13938